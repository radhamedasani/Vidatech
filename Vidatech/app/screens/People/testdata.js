
export const people = {
    "id": "1",
    "createdAt": "2019-04-29T10:04:24.713Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/johndezember/128.jpg",
    "jobTitle": "Dynamic Implementation Designer",
    "phone": "556.662.4422 x3344",
    "favouriteColor": "#023a77",
    "email": "Benny78@gmail.com",
    "firstName": "Jan",
    "lastName": "Thompson",
    "name": "Jan Thompson",
}

export const PeopleResponse = [{
    "id": "1",
    "createdAt": "2019-04-29T10:04:24.713Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/johndezember/128.jpg",
    "jobTitle": "Dynamic Implementation Designer",
    "phone": "556.662.4422 x3344",
    "favouriteColor": "#023a77",
    "email": "Benny78@gmail.com",
    "firstName": "Jan",
    "lastName": "Thompson"
}, {
    "id": "2",
    "createdAt": "2019-04-28T21:05:15.710Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/lingeswaran/128.jpg",
    "jobTitle": "Lead Response Coordinator",
    "phone": "626-316-8058 x077",
    "favouriteColor": "#39653b",
    "email": "Jordon.Parker30@hotmail.com",
    "firstName": "Rhett",
    "lastName": "Carter"
}, {
    "id": "3",
    "createdAt": "2019-04-29T10:33:47.141Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/ruehldesign/128.jpg",
    "jobTitle": "Lead Data Planner",
    "phone": "1-852-834-5571 x8035",
    "favouriteColor": "#1b366f",
    "email": "Marcus.Kuphal78@hotmail.com",
    "firstName": "Rodrigo",
    "lastName": "O'Conner"
}, {
    "id": "4",
    "createdAt": "2019-04-28T20:35:29.203Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/kolsvein/128.jpg",
    "jobTitle": "Product Factors Developer",
    "phone": "646.998.1257",
    "favouriteColor": "#6e0a75",
    "email": "Allene.Keeling@hotmail.com",
    "firstName": "Jamey",
    "lastName": "Mayer"
}]