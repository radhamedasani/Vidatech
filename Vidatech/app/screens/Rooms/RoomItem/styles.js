import { StyleSheet } from 'react-native'
import { colors, fontSize } from '../../../themes/base'

export const roomStyles = StyleSheet.create({
    container: {
        justifyContent: 'center',
        padding: 5
    },
    roomName: {
        fontSize: fontSize.md,
    },
    created: {
        fontSize: fontSize.sm,
        color: colors.teritary
    }
})