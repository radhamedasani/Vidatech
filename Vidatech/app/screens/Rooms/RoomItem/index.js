import React from 'react'
import { View, Text } from 'react-native'
import { roomStyles } from './styles'
import { formatDate } from '../../../utilities'
import { colors } from '../../../themes/base'
import { BoxContainer } from '../../../components/BoxContainer'

export const RoomItem = (props) => {
    const { name, createdAt, isOccupied } = props;
    const OccupiedColor = isOccupied ? 'red' : 'green';
    const availability = isOccupied ? 'Not Available' : 'Available';
    const occupiedTxtColor = {
        color: OccupiedColor
    }
    const { container, roomName, created } = roomStyles
    return (<BoxContainer><View style={container}>
        <Text style={roomName}>{name}</Text>
        <Text style={created}>{`Created At ${formatDate(createdAt)}`}</Text>
        <Text style={[availability, occupiedTxtColor]}>{availability}</Text>
    </View>
    </BoxContainer>)
}