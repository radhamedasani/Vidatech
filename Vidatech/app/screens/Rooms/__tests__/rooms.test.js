import React from 'react'
import reactTestRenderer from 'react-test-renderer'

import { room } from '../../testdata'
import { RoomItem, } from '../RoomItem'

describe('Test Room Components', () => {
    it('Test Room Item', () => {
        const roomItem = reactTestRenderer.create(<RoomItem {...room} />)
        expect(roomItem).toMatchSnapshot();
    })
})