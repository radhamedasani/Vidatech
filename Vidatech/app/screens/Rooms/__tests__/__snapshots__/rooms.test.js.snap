// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Room Components Test Room Item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "marginBottom": 10,
      "marginHorizontal": 10,
      "paddingLeft": 10,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      pixel
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 14,
        }
      }
    >
      Created At 29 Apr 2019
    </Text>
    <Text
      style={
        Array [
          "Available",
          Object {
            "color": "green",
          },
        ]
      }
    >
      Available
    </Text>
  </View>
</View>
`;
